app/screens/home/search/search.tsx

```diff
@@ -118,15 +130,19 @@ const SearchScreen = ({teamId}: Props) => {
         };
     }, [headerHeight, lastSearchedValue]);
 
+    const onClear = useCallback(() => {
+        setFilter('all');
```

Comment:
should we also clear the `searchValue` as well ?

Reply:
Yes, they need to be reset also. I added clearing `searchValue` and `lastSearchedValue`.  I didn't have them in this PR, but they will be handled in https://github.com/mattermost/mattermost-mobile/pull/6393 when the search modifiers screen is added. They values will be reset and the `showResults` boolean is set to false so the modifiers screen will appear on clear.

<img width="723" alt="image" src="https://user-images.githubusercontent.com/7575921/174450249-c5d7d7d5-b720-4bd8-a2dd-d277169cb0db.png">